include/dba.php

```diff
@@ -1083,15 +1084,40 @@ public static function update($table, $fields, $condition, $old_fields = array()
 		return self::e($sql, $params);
 	}
 
+	/**
+	 * Retrieve a single record from a table and returns it in an associative array
+	 * 
+	 * @brief Retrieve a single record from a table
+	 * @param string $table
+	 * @param array  $fields
+	 * @param array  $condition
+	 * @param array  $params
+	 * @return bool|array
+	 * @see dba::select
+	 */
+	public static function selectFirst($table, array $fields = [], array $condition = [], $params = [])
+	{
+		$params['limit'] = 1;
+		$result = self::select($table, $fields, $condition, $params);
+
+		if (is_bool($result)) {
+			return $result;
+		} else {
+			$row = self::fetch($result);
```

Comment:
BTW: What happens when the query is valid but there is no result?

Reply:
Same as before, it returns `false`, which means we still need `DBM::is_result`.

Reply:
The function ```dba::inArray()``` returns an empty array but no boolean value. Shouldn't we make this similar?

Means: You could change the function above to:
```return self::inArray($result, true);```

Reply:
No, `selectFirst` returns an array of fields, not rows, so if there’s no row, there shouldn’t be an array in the first place.